File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        occurences = {}
        for num in nums:
            if num in occurences.keys():
                occurences[num] = occurences[num] + 1
            else:
                occurences[num] = 1

        for value in occurences.values():
            if (value > 1):
                return True

        return False

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int product = 1;
        int numZeroes = 0;
        foreach (int num in nums) {
            if (num != 0) {
                product *= num;
            } else {
                numZeroes++;
            }
        }
        
        if (numZeroes > 1) {
            return nums.Select(n => 0).ToArray();
        }

        if (numZeroes == 1) {
            return nums.Select(n => {
                if (n == 0) {
                    return product;
                }
                return 0;
            }).ToArray();
        }

        return nums.Select(n => product / n).ToArray();
    }
}
